cloud/azure/bin/lib/bastion.sh

```diff
@@ -141,6 +141,36 @@ function bastion::get_pg_restore_command() {
       -a < ${3}"
 }
 
+#######################################
+# Get the command to dump db data.
+# Arguments:
+#   1: the postgres host to connect to
+#   2: the vaultname where secrets are stored
+#   3: path of the dump data
+#######################################
+function bastion::get_pg_dump_command() {
+  local DB_PASSWORD=$(bastion::get_pg_password "${2}")
+  echo "export DEBIAN_FRONTEND='noninteractive'; \
```

Comment:
This is a hard one, with passing it to the bastion::bastion_ssh_exec command.  Maybe another approach is:
1) Create a standalone script to install pgdump (the deb & apt-get commands)
2) scp it over to the bastion
3) Make the bastion::bastion_ssh_exec command `./install_pgdump && PGPASSWORD='${DB_PASSWORD}' /usr/lib/postgresql/14/bin/pg_dump.....`


That way this is much shorter.  

Also, I don't think the "echo" and the capturing the results is any better than just storing it in a variable.   Since this is only used by one script, who not just leave it inline, rather than pulling out into this function?